marin/post_training/load_environments.py

```diff
@@ -0,0 +1,68 @@
+import json
+from typing import Any
+
+from transformers import AutoTokenizer
+
+from .environments.marin_env import MarinEnv
+from .environments.math_env import MathEnv
+from .environments.olym_math_env import OlymMathEnv
+from .environments.swe_bench_env import SWEBenchEnv
+
+# Specify environments here
+ENVIRONMENT_NAME_TO_CLASS = {
+    "math": MathEnv,
+    "olym_math": OlymMathEnv,
+    "swe_bench": SWEBenchEnv,
+}
+
+
+def str_to_val(v: str) -> Any:
+    """Best effort convert string to bool/int/float, else keep str."""
+    v = v.strip()
+    if v.lower() in {"true", "false"}:
+        return v.lower() == "true"
+    try:
+        return int(v)
+    except ValueError:
+        pass
+    try:
+        return float(v)
+    except ValueError:
+        return v
+
+
+def load_environment_from_spec(spec: str, tokenizer: AutoTokenizer) -> MarinEnv:
+    """
+    Instantiate an environment from a spec string like 'OlymMathEnv:difficulty=easy,language=en'
```

Comment:
This logic seems pretty specific to OlymMatheEnv and may not apply to other new environments we add? Each environment will probably have different name formats and kwargs. The previous if/else logic seems more easily understandable to me. I agree we can refactor the logic for loading environments outside of train.py but maybe its worth separating the special handling of each environment and have some documentation that enumerates all configurations (e.g., ['OlymMathEnv:difficulty=easy,language=en', 'OlymMathEnv:difficulty=medium,language=en',...])?

Reply:
This logic is pretty general and can work for any args and no args environments. I also have a separate file so this is the file where people add environments.